### PR TITLE
db table migrate, rename, drop

### PR DESCRIPTION
**Pull request by earonesty**
db migration

 - rename tables
 - indexes have uuids
 - drop indexes on a table by property match
 - drop tables by name

### CODE REVIEW

File: notanorm/base.py [hunk 1]
```diff
@@ -403,14 +405,31 @@ def capture_sql(
...
+                self.drop_index_by_name(table, index.name)
+
+    def drop_index_by_name(self, table: str, index_name):
+        sql = "drop index " + self.quote_key(index_name)
+        self.execute(sql)
+
+    @staticmethod
+    def unique_index_name(table, field_names):
+        # prefer urandom to secrets, less opaque
+        return "ix_" + table + "_" + "_".join(field_names) + "_" + os.urandom(16).hex()
```
Comment (earthman1) [inline notanorm/base.py:432]: Is there a length limit to an index name that might be overrun by joining the field names together?

Comment (earthman1) [reply notanorm/base.py:432]: I think it's 62 characters

**Review by earthman1 — APPROVED** [review]

File: notanorm/base.py (hunk 1 above)
Comment (earonesty) [reply notanorm/base.py:432]: mysql 8.0 : no meaningful limit (1gb i think).  sqlite :  1000000 chars, postgres: 63 chars.  drivers can override "unique_index_name" (which has been deliberately moved to db/base.py if needed

Comment (earonesty) [reply notanorm/base.py:432]: imo: mysql 5.7 and postgres drivers should use shorter names (ix_table_uuid should be fine)

File: notanorm/base.py [hunk 2]
```diff
@@ -1153,6 +1182,21 @@ def count(self, table, where=None, **kws):
...
+        sql = (
+            "alter table "
+            + self.quote_key(table_from)
+            + " rename to "
+            + self.quote_key(table_to)
+        )
+        return self.execute(sql)
+
+    def drop(self, table):
+        """Rename a table"""
```
Comment (howufiger7) [inline notanorm/base.py:1196]: minor - update comment

File: notanorm/model.py [hunk 3]
```diff
@@ -93,6 +93,7 @@ class DbIndex(NamedTuple):
     fields: Tuple[DbIndexField, ...]
     unique: bool = False  # has a unique index?
     primary: bool = False  # is the primary key?
+    ix__name: str = None  # only avail when reading from a db model, not a ddl
```
Comment (howufiger7) [inline notanorm/model.py:96]: technically Optional[str] if it can be None

also, why use the double underscore?

**Review by howufiger7 — APPROVED** [review]
LGTM

File: notanorm/model.py (hunk 3 above)
Comment (earonesty) [reply notanorm/model.py]: so nobody uses it

File: notanorm/base.py [hunk 4]
```diff
@@ -631,14 +633,41 @@ def capture_sql(
...
+        self, name, schema: DbTable, ignore_existing=False
+    ):  # pragma: no cover
+        raise RuntimeError("Generic create table not supported")
 
-    def create_indexes(self, name, schema: DbTable, ignore_existing=False):
-        raise RuntimeError("Generic models not supported")
+    def create_indexes(self, name, schema: DbTable):  # pragma: no cover
+        raise RuntimeError("Generic create index not supported")
+
+    def drop_index(self, table: str, index: DbIndex):
```
Comment (earonesty) [inline notanorm/base.py:648]: make this lookup a separate function

**Pull request merged**